analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/HashesShouldHaveUnpredictableSaltBase.cs

```diff
@@ -18,12 +18,89 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System.Security.Cryptography;
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks;
 
 public abstract class HashesShouldHaveUnpredictableSaltBase : SymbolicRuleCheck
 {
     protected const string DiagnosticId = "S2053";
     protected const string MessageFormat = "{0}";
-    protected const string MakeSaltUnpredictableMessage = "Make this salt unpredictable.";
-    protected const string MakeThisSaltLongerMessage = "Make this salt at least 16 bytes.";
+
+    private const int SafeSaltSize = 16;
+    private const string MakeSaltUnpredictableMessage = "Make this salt unpredictable.";
+    private const string MakeThisSaltLongerMessage = "Make this salt at least 16 bytes.";
+
+    private static readonly ImmutableArray<MemberDescriptor> CryptographicallyStrongRandomNumberGenerators = ImmutableArray.Create(
+        new MemberDescriptor(KnownType.System_Security_Cryptography_RandomNumberGenerator, nameof(RandomNumberGenerator.GetBytes)),
+        new MemberDescriptor(KnownType.System_Security_Cryptography_RandomNumberGenerator, nameof(RandomNumberGenerator.GetNonZeroBytes)));
+
+    protected override ProgramState PreProcessSimple(SymbolicContext context)
+    {
+        var state = context.State;
+        var instance = context.Operation.Instance;
+        if (instance.AsObjectCreation() is { } objectCreation)
+        {
+            return ProcessObjectCreation(objectCreation, state);
+        }
+        else if (instance.AsInvocation() is { } invocation)
+        {
+            return ProcessInvocation(invocation, state);
+        }
+        else if (instance.AsArrayCreation() is { } arrayCreation)
+        {
+            return ProcessArrayCreation(arrayCreation, state);
+        }
+        else
+        {
+            return state;
+        }
+    }
+
+    private ProgramState ProcessObjectCreation(IObjectCreationOperationWrapper objectCreation, ProgramState state)
+    {
+        if (objectCreation.Type.DerivesFrom(KnownType.System_Security_Cryptography_DeriveBytes)
+            && objectCreation.Arguments.FirstOrDefault(x => x.AsArgument() is { Parameter.Name: "salt" or "rgbSalt" }) is { } saltArgument)
```

Comment:
Same here with this [comment ](https://github.com/SonarSource/sonar-dotnet/pull/7575/files#r1265397716).
Is is possible to search in the arguments with some pattern?